labelbox/data/metrics/iou.py

```diff
@@ -0,0 +1,290 @@
+    include_classifications=True
+) -> Tuple[Dict[str, List[NDAnnotation]], Dict[str, List[NDAnnotation]],
+           List[str]]:
+    """
+
+    This function takes in the raw json payloads, validates, and converts to python objects.
+    In the future datarow_miou will directly take the objects as args.
+
+    Args:
+        label_content : one row from the bulk label export - `project.export_labels()`
+        ndjsons: Model predictions in the ndjson format specified here (https://docs.labelbox.com/data-model/en/index-en#annotations)
+    Returns a tuple containing:
+        - a dict for looking up a list of predictions by feature schema id
+        - a dict for looking up a list of labels by feature schema id
+        - a list of a all feature schema ids
+
+    """
+    labels = label_content['Label'].get('objects')
+    if include_classifications:
+        labels += label_content['Label'].get('classifications')
+
+    predictions = [NDAnnotation(**pred.copy()) for pred in ndjsons]
+
+    unique_datarows = {pred.dataRow.id for pred in predictions}
+    if len(unique_datarows):
+        # Empty set of annotations is valid (if labels exist but no inferences then iou will be 0.)
+        if unique_datarows != {label_content['DataRow ID']}:
+            raise ValueError(
+                f"There should only be one datarow passed to the datarow_miou function. Found {unique_datarows}"
+            )
+
+    labels = [
+        label_to_ndannotation(label, label_content['DataRow ID'])
+        for label in labels
+    ]
+
+    labels = create_schema_lookup(labels)
+    predictions = create_schema_lookup(predictions)
+
+    feature_schemas = set(predictions.keys()).union(set(labels.keys()))
+    return predictions, labels, feature_schemas
+
+
+def datarow_miou(label_content: List[Dict[str, Any]],
+                 ndjsons: List[Dict[str, Any]],
+                 include_classifications=True,
+                 include_subclasses=True) -> float:
+    """
+
+    Args:
+        label_content : one row from the bulk label export - `project.export_labels()`
+        ndjsons: Model predictions in the ndjson format specified here (https://docs.labelbox.com/data-model/en/index-en#annotations)
+        include_classifications: Whether or not to factor top level classifications into the iou score.
+        include_subclassifications: Whether or not to factor in subclassifications into the iou score
+    Returns:
+        float indicating the iou score for this data row.
+
+    """
+
+    predictions, labels, feature_schemas = preprocess_args(
+        label_content, ndjsons, include_classifications)
+
+    ious = [
+        feature_miou(predictions[feature_schema],
+                     labels[feature_schema],
+                     include_subclasses=include_subclasses)
+        for feature_schema in feature_schemas
+    ]
+    ious = [iou for iou in ious if iou is not None]
+    if not ious:
+        return None
+    return 100 * np.mean(ious)
```

Comment:
I wasn't sure where to put this `* 100`. I kind of feel like this should just be a float that is bound between 0 and 1. Any thoughts?

Reply:
fyi I reverted this back ( remove the 100) but I still want to discuss.

Reply:
I feel better for some reason with values like this as within [0, 1].
But whatever is typical in this domain would also be a good way to decide.

Reply:
I think 0 to 1 makes the most sense